examples/read_pem_files/main.rs

```diff
@@ -0,0 +1,63 @@
+use config::{Config, Format, Map, Value, ValueKind, FileStoredFormat};
+
+/// Global settings for the exposing all preconfigured variables
+#[derive(serde::Deserialize, Clone, Debug)]
+pub struct Settings {
+    pub private_key: Option<String>,
+    pub public_key: Option<String>,
+}
+
+fn get_settings() -> Result<Settings, config::ConfigError> {
+    let settings = Config::builder()
+        // read public key
+        .add_source(config::File::new("examples/read_pem_files/cert/public.pem", PemFile).required(false))
+        // read private key
+        .add_source(config::File::new("examples/read_pem_files/cert/private.pem", PemFile).required(false))
+        .build()
+        .unwrap();
+
+    settings
+        .try_deserialize::<Settings>()
+}
+
+#[derive(Debug, Clone)]
+pub struct PemFile;
+
+impl Format for PemFile {
+    fn parse(
+        &self,
+        uri: Option<&String>,
+        text: &str,
+    ) -> Result<Map<String, config::Value>, Box<dyn std::error::Error + Send + Sync>> {
+
+        let mut result = Map::new();
+
+        let key = match text.contains("PRIVATE") {
+            true => String::from("private_key"),
+            false => String::from("public_key")
+        };
+
+        result.insert(
+            key,
+            Value::new(uri, ValueKind::String(text.into())),
+        );
+
+        Ok(result)
+    }
+}
+
+static PEM_EXT: Vec<&'static str> = vec![];
+impl FileStoredFormat for PemFile {
+    fn file_extensions(&self) -> &'static [&'static str] {
+        &PEM_EXT
+    }
+}
```

Comment:
When I first saw this in a custom format example, it was a bit confusing. That was focused on a string input only, not a file. As such this expresses no file extension associated to the format, which in this case is incorrect.

```suggestion
// A slice of extensions associated to this format, when an extension
// is omitted from a file source, these will be tried implicitly:
impl FileStoredFormat for PemFile {
    fn file_extensions(&self) -> &'static [&'static str] {
        &["pem"]
    }
}
```